### ISSUE TITLE
[Bug]: Impossible to confirm password during user creation

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#33637]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

![Bildschirmfoto vom 2022-08-22 11-10-28](https://user-images.githubusercontent.com/1374172/185886406-161cc1fc-2d71-4e0a-be3e-9cf7e9da0a8f.png)

^ this dialoge is unusable. I can't enter the password nor can I click a button.

Calling `OC.PasswordConfirmation.requirePasswordConfirmation()` after a fresh page load is usable. The confirmation dialogue only doesn't work when a modal is open.

### Steps to reproduce

1. Log into Nextcloud as admin
2. Wait x hours for the password confirmation to kick in (don't know the exact time)
3. Open the user management
4. Create a new user

### Expected behavior

Password confirmation dialogue opens and I can use it

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
N/a
```

**Comment by PVince81** [comment]
possibly related to the v5.4.0 update of nextcloud-vue, which brought the focus trap for modals: https://github.com/nextcloud/nextcloud-vue/pull/2654

**Comment by ChristophWurst** [comment]
then it's indeed the identical issue that we also had/have with Mail and v5.4 https://github.com/nextcloud/mail/pull/7034#pullrequestreview-1076893447

**Comment by PVince81** [comment]
the backports were merged, need to retry with stable5 of nextcloud-vue to see if the issue is fixed

**Comment by PVince81** [comment]
I've tested with stable5 with the fixes merged, but the problem persists

I suspect this is due to having two modals on top of each other, maybe confusing the focus trap

**Comment by PVince81** [comment]
yeah... I see that the focus is still on the modal behind and not in the new modal... you can still tab onto the fields...

<img width="226" alt="image" src="https://user-images.githubusercontent.com/277525/185898882-9551966a-6437-4e4a-9e89-47599e0eb48d.png">

**Comment by PVince81** [comment]
I tried upgrading the lib to 6.9.4 but it didn't help.

I'll try to build a test case directly in nextcloud-vue, this way we can test and debug it there

**Comment by PVince81** [comment]
I suspect that we need to detect when the current modal is not focused any more and then pause the focus trap until we return to that modal. We currently don't have the semantics to know which modal is currently on top.

**Comment by PVince81** [comment]
Hmm, I created an example in https://github.com/nextcloud/nextcloud-vue/pull/3087 and the focus trap works fine there.
Either there are more changes on master / nc vue 6.0.0 or it's the way how the dialog is invoked that is different.

Next up:
- [ ] test if bug exists on https://github.com/nextcloud/server/pull/33516

**Comment by PVince81** [comment]
another challenge:
- [ ] update nextcloud-vue again and add "Nc" prefix in calendar-availability-vue:

ERROR in ./node_modules/@nextcloud/calendar-availability-vue/lib/index.esm.js 1:0-75
Module not found: Error: Can't resolve '@nextcloud/vue/dist/Components/DatetimePicker' in '/srv/www/htdocs/server/node_modules/@nextcloud/calendar-availability-vue/lib'

**Comment by PVince81** [comment]
I forgot we already had this ticket.

Anyway, I made a new one moving forward: https://github.com/nextcloud/server/issues/33675
